pelican/contents.py

```diff
@@ -98,14 +98,18 @@ def __init__(self, content, metadata=None, settings=None,
         if not hasattr(self, 'slug'):
             if (settings['SLUGIFY_SOURCE'] == 'title' and
                     hasattr(self, 'title')):
-                self.slug = slugify(self.title,
-                                    settings.get('SLUG_SUBSTITUTIONS', ()))
+                self.slug = slugify(
+                    self.title,
+                    regex_subs=settings.get('SLUG_REGEX_SUBSTITUTIONS', []),
+                    substitutions=settings.get('SLUG_SUBSTITUTIONS'))
```

Comment:
This isn't the way we handle deprecations. If a setting is replaced with another and old one can be converted to the new one, old one is transferred to the new one in `settings.py` (see [example](https://github.com/oulenz/pelican/blob/master/pelican/settings.py#L383)). Then, old setting is no longer used in the code.

Here you can easily transfer `SLUG_SUBSTITUTIONS` to the new regex counterpart (using `re.escape` possibly).